src/test/java/com/hartwig/snpcheck/SnpCheckTest.java

```diff
@@ -216,15 +230,50 @@ public void publishesTurquoiseEventOnCompletion() throws Exception {
         fullSnpcheckWithResult(VcfComparison.Result.PASS, BARCODE);
         victim.handle(stagedEvent(Type.TERTIARY, Context.DIAGNOSTIC));
         ArgumentCaptor<PubsubMessage> pubsubMessageArgumentCaptor = ArgumentCaptor.forClass(PubsubMessage.class);
-        verify(publisher).publish(pubsubMessageArgumentCaptor.capture());
-        Map<Object, Object> message =
-                new ObjectMapper().readValue(new String(pubsubMessageArgumentCaptor.getValue().getData().toByteArray()),
-                        new TypeReference<>() {
-                        });
-        assertThat(message.get("type")).isEqualTo("snpcheck.completed");
-        List<Map<String, String>> subjects = (List<Map<String, String>>) message.get("subjects");
-        assertThat(subjects.get(0).get("name")).isEqualTo("samplet");
-        assertThat(subjects.get(0).get("type")).isEqualTo("sample");
+        verify(publisher, times(2)).publish(pubsubMessageArgumentCaptor.capture());
+        boolean foundTurquoiseEvent = false;
+        for (PubsubMessage rawMessage: pubsubMessageArgumentCaptor.getAllValues()) {
```

Comment:
this is a bit awkward. lets try to refactor out the loop and the if(keySet). Maybe using contains or something?

Reply:
See what you think of what I've just pushed, the fundamental problem is that the events are essentially maps and both a bit different in structure.